crates/sui-framework/sources/crypto.move

```diff
@@ -26,4 +28,25 @@ module sui::crypto {
     /// If the signature is a valid BLS12381 signature of the message and public key, return true.
     /// Otherwise, return false.
     public native fun bls12381_verify_g1_sig(signature: vector<u8>, public_key: vector<u8>, msg: vector<u8>): bool; 
+
+    /// @param signature: 32-byte signature that is a point on the Ed25519 elliptic curve.
+    /// @param public_key: 32-byte signature that is a point on the Ed25519 elliptic curve.
+    /// @param msg: The message that we test the signature against.
+    /// 
+    /// If the signature is a valid BLS12381 signature of the message and public key, return true.
+    /// Otherwise, return false.
+    public(friend) native fun ed25519_verify(signature: vector<u8>, public_key: vector<u8>, msg: vector<u8>): bool;
```

Comment:
looks like we more rust changes are added to this PR, are the move side of things breaking? or we can perhaps land this part without breaking genesis

Reply:
@joyqvq Yeah, we definitely need the Move side of things.

Reply:
I can definitely comment out a few lines to not make it breaking though - although the PR would literally be just 50 lines or something.